theano/sandbox/gpuarray/tests/test_abstractconv.py

```diff
@@ -75,6 +75,7 @@ def run_fwd(self, inputs_shape, filters_shape, ref=dnn_conv,
                         filter_flip=filter_flip,
                         input_shape=imshp,
                         filter_shape=kshp)
+        self.assertTrue(hasattr(c.tag, 'trace'))
         f_ref = theano.function([], c_ref, mode=mode)
         f = theano.function([], c, mode)
```

Comment:
Actually, you should check the presence of the stack trace on the output of the optimized graph, not the original variable.
The variable corresponding to the optimized output would be `f.maker.fgraph.outputs[0]`, and we want `f.maker.fgraph.outputs[0].tag.trace` to be there.